pkg/api/operators/apis/studyjob/v1alpha2/studyjob_types.go

```diff
@@ -0,0 +1,240 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//runtime "k8s.io/apimachinery/pkg/runtime"
+
+	pb "github.com/kubeflow/katib/pkg/api"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// StudyJobSpec defines the desired state of StudyJob
+type StudyJobSpec struct {
+	Owner                string                `json:"owner,omitempty"`
+	OptimizationType     OptimizationType      `json:"optimizationType,omitempty"`
+	OptimizationGoal     *float64              `json:"optimizationGoal,omitempty"`
+	ObjectiveValueName   string                `json:"objectiveValueName,omitempty"`
+	MaxSuggestionCount   int                   `json:"maxSuggestionCount,omitempty"`
```

Comment:
Would it make sense to somehow group `OptimizationType`, `OptimizationGoal`, `ObjectiveValueName`, and maybe `MaxSuggestionCount` and or `ObservationBudget` proposed by @jdplatt ? Could we have an Optimization/Objective Spec, so that the generic categories a user needs to define are clear and separated in the `StudyJobSpec`? Same way we have a `SuggestionSpec` for example.

Also, following [this converstation](https://kubeflow.slack.com/archives/CE0BURK1B/p1549135874084100?thread_ts=1548816849.064800&cid=CE0BURK1B) does it make sense to rename:

RequestNumber -> ParallelWorkersNum
RequestCount -> MaxIterations (rather than MaxSuggestionCount)

I think that `MaxSuggestionCount` can be confused with the max number of the aggregated suggestions one wants, rather than the actual iterations we want to do. What do you think?

Reply:
This is a great suggestion.